src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/ReaderBigEndian.cs

```diff
@@ -96,10 +124,25 @@ public static ulong ReadUInt64BigEndian(ReadOnlySpan<byte> source)
             return result;
         }
 
+        /// <summary>
+        /// Reads a Double out of a read-only span of bytes as big endian.
+        /// </summary>
+        /// <returns>If the span is too small to contain a Double, return false.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryReadDoubleBigEndian(ReadOnlySpan<byte> source, out double value)
+        {
+            bool success = MemoryMarshal.TryRead(source, out value);
+            if (BitConverter.IsLittleEndian)
+            {
+                value = ReverseEndianness(value);
+            }
+            return success;
+        }
+
         /// <summary>
         /// Reads an Int16 out of a read-only span of bytes as big endian.
-        /// <returns>If the span is too small to contain an Int16, return false.</returns>
         /// </summary>
+        /// <returns>If the span is too small to contain an Int16, return false.</returns>
```

Comment:
Thanks for catching these. :)

Reply:
Unfortunately, it doesn't look like the docs are generated from this source. The stuff at https://docs.microsoft.com/en-us/dotnet/api/system.buffers.binary.binaryprimitives.trywriteint16bigendian is much better than what we have here.

Reply:
The docs commented started with the comments from source but then went throw doc review and got fixed up (see https://github.com/dotnet/dotnet-api-docs/pull/2693). For new APIs, if you fix them in source, it will make porting them easier.

cc @carlossanlop

Reply:
Why don't the fixes from the doc review get ported back into the source?

Reply:
We do not have a automated tooling to do that. If we did have one, it would likely require a lot of manual interventions and be expensive to keep running on ongoing basis.

IMHO, the best way to address this would be to make the docs live in just one place, and avoid the duplication.

Reply:
> the best way to address this would be to make the docs live in just one place, and avoid the duplication.

Totally agree. IMO that one place should be in the source.

Reply:
> that one place should be in the source.

It is not obvious that it would be the right choice. The documentation is version-agnostic (spans all supported versions of .NET Framework and .NET Core), but the sources are version specific.

Reply:
We had similar discussions previously with the docs team and given the constraints, decided as having the docs site be the single source of truth. It does have the side effect that you are seeing @eerhardt, but as @jkotas mentioned, trying to keep source and docs in sync is challenging, potentially flaky and may not be a tenable, especially since both the docs and source get updated all the time (from various parties). Given the real customer value is having the docs and Intellisense, we prioritized that as the source of truth and only "seed" them with the xml comments from source once. The higher order bit is to encourage and ensure that we are writing useful, customer facing xml comments that are complete, whenever we add new APIs and make sure the customer benefits from it. Area owners and those adding API/source contribution _can_ manually sync them back if they choose to, but it isn't mandatory.